pyartcd/pyartcd/pipelines/update_golang.py

```diff
@@ -186,33 +182,42 @@ async def run(self):
-                                 f'Once it is available, untag it from the override tag (brew untag-build {tag} {nvr})')
-
-        # if requested, create ticket for tagging request
-        if self.create_ticket:
-            _LOGGER.info('Making sure that builds are not already tagged')
-            for el_v, nvr in el_nvr_map.items():
-                if is_latest_build(self.ocp_version, el_v, nvr, self.koji_session):
-                    raise ValueError(f'{nvr} is already the latest build, run '
-                                     'only with nvrs that are not latest or run without --create-tagging-ticket')
-
-            _LOGGER.info('Creating Jira ticket to tag golang builds in buildroots')
-            self.create_jira_ticket_for_el8(el_nvr_map, go_version)
-            return
-
         cannot_proceed = False
         for el_v, nvr in el_nvr_map.items():
+            if await is_latest_and_available(self.ocp_version, el_v, nvr, self.koji_session):
+                continue
+            if not self.tag_builds:
+                cannot_proceed = True
+                continue
+            # tag builds into override tag
+            if not self.koji_session.logged_in:
+                _LOGGER.info("user logged out from session, login again")
+                self.koji_session.gssapi_login()
+            if el_v == 8:
+                rhel8_module_tag = self.get_module_tag(nvr, el_v)
+                latest_rhel8_builds = self.koji_session.listTagged(rhel8_module_tag, latest=True, inherit=True)
+                # need to tag delve go-toolset golang 3 module builds
+                for el8_nvr in [b['nvr'] for b in latest_rhel8_builds]:
+                    self.koji_session.tagBuild(f'rhaos-{self.ocp_version}-rhel-{el_v}-override', el8_nvr)
+                    _LOGGER.info(f"Tagged {el8_nvr} with rhaos-{self.ocp_version}-rhel-{el_v}-override tag")
+            elif el_v == 9:
+                self.koji_session.tagBuild(f'rhaos-{self.ocp_version}-rhel-{el_v}-override', nvr)
+                _LOGGER.info(f"Tagged {nvr} with rhaos-{self.ocp_version}-rhel-{el_v}-override tag")
+            # need to regen-repo
+            build_tag = f'rhaos-{self.ocp_version}-rhel-{el_v}-build'
+            task_id = self.koji_session.newRepo(build_tag)
+            _LOGGER.info(f"Regenerating repo for tag: {build_tag} with task {task_id}, waiting max to 10 mins...")
+            # regen repo usually take a few mins, wait max to 10 mins
+            wait_min = 0
+            while wait_min < 10:
+                time.sleep(60)
+                gen_repo_task_info = self.koji_session.getTaskInfo(task_id)
+                if gen_repo_task_info['completion_time']:
+                    _LOGGER.info(f"Regenerating repo for tag: {build_tag} complete.")
+                    break
+                _LOGGER.info(f"Regenerating repo for tag: {build_tag} task {task_id} is still running")
+                wait_min += 1
```

Comment:
This would also be good in a separate func

Reply:
updated